google/cloud/sql/connector/connector.py

```diff
@@ -137,17 +166,21 @@ def connect(
             )
         else:
             ip_type = kwargs.pop("ip_type", self._ip_type)
-        if "timeout" in kwargs:
-            return icm.connect(driver, ip_type, **kwargs)
-        elif "connect_timeout" in kwargs:
+        timeout = kwargs.pop("timeout", self._timeout)
+        if "connect_timeout" in kwargs:
             timeout = kwargs["connect_timeout"]
-        else:
-            timeout = self._timeout
+
         try:
-            return icm.connect(driver, ip_type, timeout, **kwargs)
+            connection_task = self._loop.create_task(
+                icm.connect(driver, ip_type, **kwargs)
+            )
```

Comment:
I don't think this needs to be a task anymore since we are already in async land by this point. You can await for directly on the coro generated by calling the function.